articles/WideStrings.md

```diff
@@ -0,0 +1,264 @@
+as this is disallowed by the DDS specification.
+
+## Multi-byte character background
+
+Before delving into ROS 2.0 specifics, some time should be spent discussing multi-byte
+characters in general.  What is meant by multi-byte characters?  The original ASCII
+character set only provided for printable characters using binary sequences 0 to 127.
+Several extensions were made to this to use 128-255, but that approach only extended the
+character set to be able to cover a few more languages.  Several other encodings were
+created as the need arose, and it became clear by the 1990's that a worldwide standard
+was required.  Thus, Unicode was created to be a single specification that could cover
+all of the languages of the world.  The early versions of Unicode actually failed to
+live up to this expectation, and some of those early versions of Unicode are still
+baked into widely used software (Microsoft Windows, Java, etc).  Later on, the Unicode
+standard evolved to truly be able to cover the languages of the entire world, and this
+is what should generally be used for new designs.  As a side benefit, the newer versions of
+Unicode (UTF-8) are backwards compatible with ASCII.  There is a lot of history here, and
+things are much more complicated then the brief introduction above.  For more information,
+the following links provide introductory material:
+
+* [http://kunststube.net/encoding/](http://kunststube.net/encoding/)
+* [http://stackoverflow.com/questions/4588302/why-isnt-wchar-t-widely-used-in-code-for-linux-related-platforms](http://stackoverflow.com/questions/4588302/why-isnt-wchar-t-widely-used-in-code-for-linux-related-platforms)
+* [http://www.diveintopython3.net/strings.html](http://www.diveintopython3.net/strings.html)
+* [http://stackoverflow.com/questions/402283/stdwstring-vs-stdstring](http://stackoverflow.com/questions/402283/stdwstring-vs-stdstring)
+
+The rest of this document is generally going to assume Unicode, unless stated otherwise.
+
+## Introducing Wide Strings
+To support multi-byte characters, ROS 2.0 will introduce the concept of a wide string.  There are a
+few basic questions to answer on how a wide string will be integrated into ROS 2.0:
+
+* What is the size impact of the wide string?
+* What is the encoding of the wide string?
+* What does the API look like to a user of ROS 2.0?
+
+Each of these questions will be examined in more detail below.
+
+### What is the size impact of the wide string?
+
+There are 2 parts to this question.  One is how the wide string is mapped onto the underlying
+DDS implementation, which defines how large messages will be on-the-wire (and on disk).  The
+other part of the question deals with the runtime impact (in terms of speed and code size) of
+having to deal with wide strings.
+
+**Mapping onto the DDS implementation**
+
+There are a couple of different ways that a wide string could be mapped onto a
+DDS implementation.  The first way it could be mapped is onto a uint8_t byte
```

Comment:
overlapping @dirk-thomas comment's above, I think in this section we should refer to the DDS types like you do below when referring to the X-types spec